wagtail/admin/views/generic/models.py

```diff
@@ -256,16 +257,39 @@ def get_buttons(column, instance, *args, **kwargs):
         )
 
     def _get_custom_column(self, field_name, column_class=Column, **kwargs):
-        label, attr = label_for_field(field_name, self.model, return_attr=True)
+        *relation, field = (
+            [field_name]
+            if hasattr(self.model, field_name)
+            else field_name.split(LOOKUP_SEP)
+        )
+        model_class = self.model
+
+        # iterate over the relation list to try to get the last model
+        # where the field exists
+        foreign_field_name = ""
+        for model in relation:
+            foreign_field = model_class._meta.get_field(model)
+            foreign_field_name = foreign_field.verbose_name
+            model_class = foreign_field.related_model
+
+        label, attr = label_for_field(field, model_class, return_attr=True)
+        if foreign_field_name:
+            label = _("%(related_model_name)s %(field_label)s") % {
+                "related_model_name": foreign_field_name,
+                "field_label": label,
+            }
+
         sort_key = getattr(attr, "admin_order_field", None)
 
         # attr is None if the field is an actual database field,
         # so it's possible to sort by it
         if attr is None:
             sort_key = field_name
 
+        # build the dotted relation if needed, for the use in multigetattr
+        dotted_field = ".".join(relation + [field])
```

Comment:
If we didn't immediately unpack the splitting result into `*relation, field`, we can reuse it here so we don't have to construct a new list. It's a very small optimisation but I think it's nice to have